frontend/app/filter.tsx

```diff
@@ -0,0 +1,143 @@
+    ]);
+
+    const [selectedSort, setSelectedSort] = useState<TSortOption[]>([
+        { id: "Taste Rating", direction: "none" },
+        { id: "Value Rating", direction: "none" },
+        { id: "Portion Rating", direction: "none" },
+        { id: "Overall Rating", direction: "none" },
+    ]);
+
+    const cycleSelectedSort = (title: string) => {
+        setSelectedSort((prevSort) => {
+            return prevSort.map((item) => {
+                if (item.id === title) {
+                    // Cycle through directions: "none" -> "up" -> "down" -> "none"
+                    switch (item.direction) {
+                        case "none":
+                            // console.log(`Changing ${item.id} from none to up`);
+                            return { ...item, direction: "up" };
+                        case "up":
+                            // console.log(`Changing ${item.id} from up to down`);
+                            return { ...item, direction: "down" };
+                        case "down":
+                            // console.log(`Changing ${item.id} from down to none`);
+                            return { ...item, direction: "none" };
+                    }
+                }
+                // console.log(`Selected sort:`, item);
+
+                return { ...item, direction: "none" }; // Set all others to "none"
+            });
+        });
+    };
+
+    const toggleTag = (id: TFilterId, setTags: React.Dispatch<React.SetStateAction<TFilterItem[]>>) => {
+        setTags((prevTags) => {
+            const updated = prevTags.map((t) => (t.id === id ? { ...t, selected: !t.selected } : t));
+
+            // const selectedTags = updated.filter((t) => t.selected);
+            // console.log(`Selected tags:`, selectedTags);
+
+            return updated;
+        });
+    };
+
+    const toggleCuisineTag = (id: TFilterId) => toggleTag(id, setCuisineTags);
+    const toggleSpecificationTag = (id: TFilterId) => toggleTag(id, setSpecificationTags);
+    return (
+        // <SafeAreaView style={styles.container}>
+        <ScrollView style={{ flex: 1, backgroundColor: "#FFF" }}>
+            <ThemedView style={styles.container}>
+                <SortBy sortOptions={selectedSort} onSortOptionPress={cycleSelectedSort} />
+                <FilterGrid filters={cuisineTags} title="🍝 Cuisines" onTagPress={toggleCuisineTag} />
+                <FilterGrid filters={specificationTags} title="🍽️ Specifications" onTagPress={toggleSpecificationTag} />
+                <Button
+                    title={"Apply Filters"}
+                    onPress={() => {}}
+                    containerStyle={styles.applyButtonContainer}
+                    textStyle={styles.applyButtonText}
+                />
+            </ThemedView>
+        </ScrollView>
+
+        // </SafeAreaView>
+    );
+}
+const styles = StyleSheet.create({
+    container: {
+        flex: 1,
+        alignItems: "center",
+        paddingTop: 56,
```

Comment:
I dont know how exactly the figma laid these out but I'm wondering if theres a nicer way we can do this without exact numbers and through alignment and auto stuff - these numbers should also be in multiples of 4 (which is probably a figma issue)